### DISCUSSION TITLE
SOLVED Convert a series of .jpg's into .png's and removing background with -transparent-color #000000

### DISCUSSION THREAD

**Discussion by creinemann** [ImageMagick/ImageMagick#2709] [Help] [answered]
I am trying to add a line to my script to convert a large number of jpgs. to ,pngs. while making the background color which is black  #000000, transparent.
I am running this in a windows batch file, and cannot seem to get the correct syntax correct to get it to run without errors
this is what I have:
`echo

for /R "F:\Satellitetest\single\" %%f in (*.jpg) do "C:\Program Files\ImageMagick-7.0.9-Q16\magick.exe" convert "%%f" -transparent-color #000000 "F:\Satellitetest\single\*.png"
pause`

**Comment by creinemann** [comment, marked as answer]
SOLVED 
echo

for /R "F:\Satellitetest\single\" %%a in (*.jpg) do "C:\Program Files\ImageMagick-7.0.9-Q16\magick.exe"  mogrify -path ../Satellitetest/single/test2 -format png -transparent black -fuzz 2%%  "%%~a"

pause

**Comment by snibgo** [comment]
That is more complex than it needs to be. "magick mogrify" will loop through all its input files, so you don't need the shell "for" loop.
```
magick.exe mogrify -path ../Satellitetest/single/test2 -format png -transparent black -fuzz 2%% *.jpg
```

**Reply by creinemann** [reply]
Thanks, I'll try it out!

**Reply by creinemann** [reply]
Just had to change it to this:
`magick.exe mogrify -path ../Satellitetest/single/test2 -format png -transparent black -fuzz 2%% F:\Satellitetest\single\*.jpg`
